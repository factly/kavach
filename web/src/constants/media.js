//Actions
export const ADD_MEDIUM = 'ADD_MEDIUM';
export const ADD_MEDIA = 'ADD_MEDIA';
export const ADD_MEDIA_REQUEST = 'ADD_MEDIA_REQUEST';
export const RESET_MEDIA = 'RESET_MEDIA';
export const SET_MEDIA_LOADING = 'SET_MEDIA_LOADING';

//API
export const MEDIA_API = '/media';
