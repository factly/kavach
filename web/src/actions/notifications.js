import { ADD_NOTIFICATION } from '../constants/notifications';

export const addErrorNotification = (data) => ({
  type: ADD_NOTIFICATION,
  payload: {
    type: 'error',
    title: 'Error',
    message: data,
  },
});

export const addSuccessNotification = (data) => ({
  type: ADD_NOTIFICATION,
  payload: {
    type: 'success',
    title: 'Success',
    message: data,
  },
});
