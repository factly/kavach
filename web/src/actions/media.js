import axios from 'axios';
import {
  ADD_MEDIUM,
  ADD_MEDIA,
  ADD_MEDIA_REQUEST,
  SET_MEDIA_LOADING,
  RESET_MEDIA,
  MEDIA_API,
} from '../constants/media';
import { buildObjectOfItems } from '../utils/objects';

import { addErrorNotification, addSuccessNotification } from './notifications';

export const getMedia = (query) => {
  return (dispatch) => {
    dispatch(loadingMedia());
    return axios
      .get(MEDIA_API, {
        params: query,
      })
      .then((response) => {
        dispatch(addMediaList(response.data.nodes));
        dispatch(
          addMediaRequest({
            data: response.data.nodes.map((item) => item.id),
            query: query,
            total: response.data.total,
          }),
        );
        dispatch(stopMediaLoading());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const getMedium = (id) => {
  return (dispatch) => {
    dispatch(loadingMedia());
    return axios
      .get(MEDIA_API + '/' + id)
      .then((response) => {
        dispatch(getMediumByID(response.data));
        dispatch(stopMediaLoading());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const addMedium = (data) => {
  return (dispatch) => {
    dispatch(loadingMedia());
    return axios
      .post(MEDIA_API, data)
      .then(() => {
        dispatch(resetMedia());
        dispatch(addSuccessNotification('Media Added'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const updateMedium = (data) => {
  return (dispatch) => {
    dispatch(loadingMedia());
    return axios
      .put(MEDIA_API + '/' + data.id, data)
      .then((response) => {
        dispatch(getMediumByID(response.data));
        dispatch(stopMediaLoading());
        dispatch(addSuccessNotification('Media Updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const deleteMedium = (id) => {
  return (dispatch) => {
    dispatch(loadingMedia());
    return axios
      .delete(MEDIA_API + '/' + id)
      .then(() => {
        dispatch(resetMedia());
        dispatch(addSuccessNotification('Media Deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const addMedia = (media) => {
  return (dispatch) => {
    dispatch(addMediaList(media));
  };
};

export const loadingMedia = () => ({
  type: SET_MEDIA_LOADING,
  payload: true,
});

export const stopMediaLoading = () => ({
  type: SET_MEDIA_LOADING,
  payload: false,
});

export const getMediumByID = (data) => ({
  type: ADD_MEDIUM,
  payload: data,
});

export const addMediaList = (data) => (dispatch) => {
  dispatch({
    type: ADD_MEDIA,
    payload: buildObjectOfItems(data),
  });
};

export const addMediaRequest = (data) => ({
  type: ADD_MEDIA_REQUEST,
  payload: data,
});

export const resetMedia = () => ({
  type: RESET_MEDIA,
});
