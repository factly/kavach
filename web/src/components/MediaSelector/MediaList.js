import React from 'react';
import { List, Avatar, Space, Input } from 'antd';
import { useDispatch, useSelector } from 'react-redux';
import { getMedia } from '../../actions/media';
import deepEqual from 'deep-equal';
import { CheckCircleTwoTone } from '@ant-design/icons';

function MediaList({ onSelect, selected, onUnselect }) {
  const dispatch = useDispatch();

  const [filters, setFilters] = React.useState({
    page: 1,
    limit: 8,
  });

  const { media, total } = useSelector((state) => {
    const node = state.media.req.find((item) => {
      return deepEqual(item.query, filters);
    });

    if (node)
      return {
        media: node.data.map((element) => state.media.details[element]),
        total: node.total,
      };

    return { media: [], total: 0 };
  });

  React.useEffect(() => {
    fetchMedia();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dispatch, filters]);

  const fetchMedia = () => {
    dispatch(getMedia(filters));
  };

  return (
    <Space direction={'vertical'}>
      <Input
        placeholder="Search Media"
        onChange={(e) =>
          e.target.value ? setFilters({ ...filters, q: e.target.value }) : setFilters(filters)
        }
      />
      <List
        grid={{
          gutter: 16,
          md: 4,
        }}
        pagination={{
          current: filters.page,
          total: total,
          pageSize: filters.limit,
          onChange: (pageNumber, pageSize) => {
            setFilters({ page: pageNumber, limit: pageSize });
          },
        }}
        dataSource={media}
        renderItem={(item) => (
          <List.Item>
            {selected && item.id === selected.id ? (
              <div style={{ position: 'relative' }}>
                <Avatar
                  onClick={() => {
                    onSelect(null);
                    onUnselect();
                  }}
                  shape="square"
                  size={174}
                  src={(window.REACT_APP_ENABLE_IMGPROXY) ? item.url?.proxy : item.url?.raw}
                  style={{ opacity: '0.7', padding: '0.5rem', border: '2px solid #1890ff' }}
                />
                <CheckCircleTwoTone
                  twoToneColor="#52c41a"
                  style={{ fontSize: '2.5rem', position: 'absolute', top: 8, right: 8 }}
                />
              </div>
            ) : (
              <Avatar
                onClick={() => onSelect(item)}
                shape="square"
                size={174}
                src={(window.REACT_APP_ENABLE_IMGPROXY) ? item.url?.proxy : item.url?.raw}
              />
            )}
          </List.Item>
        )}
      />
    </Space>
  );
}

export default MediaList;
