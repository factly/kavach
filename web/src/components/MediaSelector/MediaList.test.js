import React from 'react';
import { Provider } from 'react-redux';
import { useDispatch } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '../../matchMedia.mock';
import MediaList from './MediaList';
import { mount } from 'enzyme';
import { List } from 'antd';
import { act } from 'react-dom/test-utils';
import * as actions from '../../actions/media';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));
jest.mock('../../actions/media', () => ({
  getMedia: jest.fn(),
}));
let state = {
  media: {
    req: [
      {
        data: [1, 2],
        query: { page: 1, limit: 8 },
        total: 2,
      },
    ],
    details: {
      1: {
        id: 1,
        name: 'Medium -1',
        url: 'some-url',
        file_size: 'file_size',
        caption: 'caption',
        description: 'description',
      },
      2: {
        id: 2,
        name: 'Medium - 2',
        url: 'some-url',
        file_size: 'file_size',
        caption: 'caption',
        description: 'description',
      },
    },
    loading: false,
  },
};

describe('Media List component', () => {
  let store;
  let mockedDispatch;

  describe('snapshot testing', () => {
    beforeEach(() => {
      store = mockStore(() => {});
      store.dispatch = jest.fn();
      mockedDispatch = jest.fn();
      useDispatch.mockReturnValue(mockedDispatch);
    });

    it('should match component with data', () => {
      store = mockStore(state);
      const tree = mount(
        <Provider store={store}>
          <MediaList selected={{ id: 1 }} />
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    beforeEach(() => {
      jest.clearAllMocks();
      mockedDispatch = jest.fn(() => new Promise((resolve) => resolve(true)));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should change the page', () => {
      actions.getMedia.mockReset();
      store = mockStore(() => state);
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <MediaList />
          </Provider>,
        );
      });

      const list = wrapper.find(List);
      list.props().pagination.onChange(2);
      wrapper.update();
      const updatedTable = wrapper.find(List);
      expect(updatedTable.props().pagination.current).toEqual(2);
      expect(actions.getMedia).toHaveBeenCalledWith({ page: 1, limit: 8 });
    });
    test('should handle rendering avatar when IMGPROXY is enabled', () => {
      store = mockStore({
        ...state,
        media: {
          ...state.media,
          details: {
            1: {
              id: 1,
              name: 'Medium -1',
              url: {
                proxy: "example proxy url",
                raw:  "example raw url",
              },
              file_size: 'file_size',
              caption: 'caption',
              description: 'description',
            },
            2: {
              id: 2,
              name: 'Medium - 2',
              url: {
                proxy: "example proxy url",
                raw:  "example raw url",
              },
              file_size: 'file_size',
              caption: 'caption',
              description: 'description',
            },
          },
        },
      });
      window.REACT_APP_ENABLE_IMGPROXY = true;
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <MediaList />
          </Provider>,
        );
      });
      expect(wrapper.find('Avatar').length).toBe(2);
      expect(wrapper.find('Avatar').at(0).props().src).toBe('example proxy url');
      expect(wrapper.find('Avatar').at(1).props().src).toBe('example proxy url');
    });
    test('should handle rendering avatar when IMGPROXY is disabled', () => {
      store = mockStore({
        ...state,
        media: {
          ...state.media,
          details: {
            1: {
              id: 1,
              name: 'Medium -1',
              url: {
                proxy: "example proxy url",
                raw:  "example raw url",
              },
              file_size: 'file_size',
              caption: 'caption',
              description: 'description',
            },
            2: {
              id: 2,
              name: 'Medium - 2',
              url: {
                proxy: "example proxy url",
                raw:  "example raw url",
              },
              file_size: 'file_size',
              caption: 'caption',
              description: 'description',
            },
          },
        },
      });
      window.REACT_APP_ENABLE_IMGPROXY = false;
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <MediaList />
          </Provider>,
        );
      });
      expect(wrapper.find('Avatar').length).toBe(2);
      expect(wrapper.find('Avatar').at(0).props().src).toBe('example raw url');
      expect(wrapper.find('Avatar').at(1).props().src).toBe('example raw url');
    });
    it('should handle image selection', () => {
      store = mockStore(state);
      const onSelect = jest.fn();
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <MediaList onSelect={onSelect} />
          </Provider>,
        );
      });
      act(() => {
        // wrapper.find('Input').simulate('change', { target: { value: '' } });
        const Input = wrapper.find('input');
        Input.simulate('change', { target: { value: 'Medium -1' } });
      });
      wrapper.find('Avatar').at(0).simulate('click');
      expect(onSelect).toHaveBeenCalledWith({
        id: 1,
        name: 'Medium -1',
        url: 'some-url',
        file_size: 'file_size',
        caption: 'caption',
        description: 'description',
      });
    });

    it('should handle image unselection', () => {
      store = mockStore(state);
      const onSelect = jest.fn();
      const onUnselect = jest.fn();

      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <MediaList
              onSelect={onSelect}
              onUnselect={onUnselect}
              selected={{
                id: 1,
                name: 'Medium -1',
                url: 'some-url',
                file_size: 'file_size',
                caption: 'caption',
                description: 'description',
              }}
            />
          </Provider>,
        );
      });
      act(() => {
        // wrapper.find('Input').simulate('change', { target: { value: '' } });
        const Input = wrapper.find('input');
        Input.simulate('change', { target: { value: '' } });
      });
      wrapper.find('Avatar').at(0).simulate('click');
      expect(onSelect).toHaveBeenCalledWith(null);
    });
  });

});
