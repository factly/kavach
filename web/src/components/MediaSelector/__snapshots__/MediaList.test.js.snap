// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media List component snapshot testing should match component with data 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MediaList
    selected={
      Object {
        "id": 1,
      }
    }
  >
    <Space
      direction="vertical"
    >
      <div
        className="ant-space ant-space-vertical"
        style={Object {}}
      >
        <Item
          className="ant-space-item"
          direction="vertical"
          index={0}
          key="ant-space-item-0"
          marginDirection="marginRight"
          wrap={false}
        >
          <div
            className="ant-space-item"
            style={
              Object {
                "marginBottom": 8,
              }
            }
          >
            <ForwardRef
              onChange={[Function]}
              placeholder="Search Media"
            >
              <ForwardRef
                affixWrapperClassName=""
                className=""
                groupClassName=""
                inputClassName=""
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Search Media"
                prefixCls="ant-input"
                wrapperClassName=""
              >
                <BaseInput
                  affixWrapperClassName=""
                  className=""
                  focused={false}
                  groupClassName=""
                  handleReset={[Function]}
                  inputElement={
                    <input
                      className="ant-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Search Media"
                      type="text"
                    />
                  }
                  placeholder="Search Media"
                  prefixCls="ant-input"
                  suffix={null}
                  triggerFocus={[Function]}
                  value=""
                  wrapperClassName=""
                >
                  <input
                    className="ant-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Search Media"
                    type="text"
                    value=""
                  />
                </BaseInput>
              </ForwardRef>
            </ForwardRef>
          </div>
        </Item>
        <Item
          className="ant-space-item"
          direction="vertical"
          index={1}
          key="ant-space-item-1"
          marginDirection="marginRight"
          wrap={false}
        >
          <div
            className="ant-space-item"
            style={Object {}}
          >
            <List
              dataSource={
                Array [
                  Object {
                    "caption": "caption",
                    "description": "description",
                    "file_size": "file_size",
                    "id": 1,
                    "name": "Medium -1",
                    "url": "some-url",
                  },
                  Object {
                    "caption": "caption",
                    "description": "description",
                    "file_size": "file_size",
                    "id": 2,
                    "name": "Medium - 2",
                    "url": "some-url",
                  },
                ]
              }
              grid={
                Object {
                  "gutter": 16,
                  "md": 4,
                }
              }
              pagination={
                Object {
                  "current": 1,
                  "onChange": [Function],
                  "pageSize": 8,
                  "total": 2,
                }
              }
              renderItem={[Function]}
            >
              <div
                className="ant-list ant-list-split ant-list-grid ant-list-something-after-last-item"
              >
                <Spin
                  spinning={false}
                >
                  <Spin
                    spinPrefixCls="ant-spin"
                    spinning={false}
                  >
                    <div
                      className="ant-spin-nested-loading"
                    >
                      <div
                        className="ant-spin-container"
                        key="container"
                      >
                        <Row
                          gutter={16}
                        >
                          <div
                            className="ant-row"
                            style={
                              Object {
                                "marginLeft": -8,
                                "marginRight": -8,
                              }
                            }
                          >
                            <div
                              key=".$list-item-0"
                            >
                              <ForwardRef(InternalItem)>
                                <Col
                                  flex={1}
                                >
                                  <div
                                    className="ant-col"
                                    style={
                                      Object {
                                        "flex": "1 1 auto",
                                        "paddingLeft": 8,
                                        "paddingRight": 8,
                                      }
                                    }
                                  >
                                    <div
                                      className="ant-list-item"
                                    >
                                      <div
                                        style={
                                          Object {
                                            "position": "relative",
                                          }
                                        }
                                      >
                                        <Avatar
                                          onClick={[Function]}
                                          shape="square"
                                          size={174}
                                          style={
                                            Object {
                                              "border": "2px solid #1890ff",
                                              "opacity": "0.7",
                                              "padding": "0.5rem",
                                            }
                                          }
                                        >
                                          <span
                                            className="ant-avatar ant-avatar-square"
                                            onClick={[Function]}
                                            style={
                                              Object {
                                                "border": "2px solid #1890ff",
                                                "fontSize": 18,
                                                "height": 174,
                                                "lineHeight": "174px",
                                                "opacity": "0.7",
                                                "padding": "0.5rem",
                                                "width": 174,
                                              }
                                            }
                                          >
                                            <ResizeObserver
                                              onResize={[Function]}
                                            >
                                              <SingleObserver
                                                key="rc-observer-key-0"
                                                onResize={[Function]}
                                              >
                                                <DomWrapper>
                                                  <span
                                                    className="ant-avatar-string"
                                                    style={
                                                      Object {
                                                        "WebkitTransform": "scale(1) translateX(-50%)",
                                                        "lineHeight": "174px",
                                                        "msTransform": "scale(1) translateX(-50%)",
                                                        "transform": "scale(1) translateX(-50%)",
                                                      }
                                                    }
                                                  />
                                                </DomWrapper>
                                              </SingleObserver>
                                            </ResizeObserver>
                                          </span>
                                        </Avatar>
                                        <ForwardRef(CheckCircleTwoTone)
                                          style={
                                            Object {
                                              "fontSize": "2.5rem",
                                              "position": "absolute",
                                              "right": 8,
                                              "top": 8,
                                            }
                                          }
                                          twoToneColor="#52c41a"
                                        >
                                          <AntdIcon
                                            icon={
                                              Object {
                                                "icon": [Function],
                                                "name": "check-circle",
                                                "theme": "twotone",
                                              }
                                            }
                                            style={
                                              Object {
                                                "fontSize": "2.5rem",
                                                "position": "absolute",
                                                "right": 8,
                                                "top": 8,
                                              }
                                            }
                                            twoToneColor="#52c41a"
                                          >
                                            <span
                                              aria-label="check-circle"
                                              className="anticon anticon-check-circle"
                                              role="img"
                                              style={
                                                Object {
                                                  "fontSize": "2.5rem",
                                                  "position": "absolute",
                                                  "right": 8,
                                                  "top": 8,
                                                }
                                              }
                                            >
                                              <IconReact
                                                icon={
                                                  Object {
                                                    "icon": [Function],
                                                    "name": "check-circle",
                                                    "theme": "twotone",
                                                  }
                                                }
                                                primaryColor="#52c41a"
                                              >
                                                <svg
                                                  aria-hidden="true"
                                                  data-icon="check-circle"
                                                  fill="currentColor"
                                                  focusable="false"
                                                  height="1em"
                                                  key="svg-check-circle"
                                                  viewBox="64 64 896 896"
                                                  width="1em"
                                                >
                                                  <path
                                                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                                                    fill="#52c41a"
                                                    key="svg-check-circle-svg-0"
                                                  />
                                                  <path
                                                    d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm193.4 225.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.3 0 19.9 5 25.9 13.3l71.2 98.8 157.2-218c6-8.4 15.7-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.4 12.7z"
                                                    fill="#f6ffed"
                                                    key="svg-check-circle-svg-1"
                                                  />
                                                  <path
                                                    d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                                                    fill="#52c41a"
                                                    key="svg-check-circle-svg-2"
                                                  />
                                                </svg>
                                              </IconReact>
                                            </span>
                                          </AntdIcon>
                                        </ForwardRef(CheckCircleTwoTone)>
                                      </div>
                                    </div>
                                  </div>
                                </Col>
                              </ForwardRef(InternalItem)>
                            </div>
                            <div
                              key=".$list-item-1"
                            >
                              <ForwardRef(InternalItem)>
                                <Col
                                  flex={1}
                                >
                                  <div
                                    className="ant-col"
                                    style={
                                      Object {
                                        "flex": "1 1 auto",
                                        "paddingLeft": 8,
                                        "paddingRight": 8,
                                      }
                                    }
                                  >
                                    <div
                                      className="ant-list-item"
                                    >
                                      <Avatar
                                        onClick={[Function]}
                                        shape="square"
                                        size={174}
                                      >
                                        <span
                                          className="ant-avatar ant-avatar-square"
                                          onClick={[Function]}
                                          style={
                                            Object {
                                              "fontSize": 18,
                                              "height": 174,
                                              "lineHeight": "174px",
                                              "width": 174,
                                            }
                                          }
                                        >
                                          <ResizeObserver
                                            onResize={[Function]}
                                          >
                                            <SingleObserver
                                              key="rc-observer-key-0"
                                              onResize={[Function]}
                                            >
                                              <DomWrapper>
                                                <span
                                                  className="ant-avatar-string"
                                                  style={
                                                    Object {
                                                      "WebkitTransform": "scale(1) translateX(-50%)",
                                                      "lineHeight": "174px",
                                                      "msTransform": "scale(1) translateX(-50%)",
                                                      "transform": "scale(1) translateX(-50%)",
                                                    }
                                                  }
                                                />
                                              </DomWrapper>
                                            </SingleObserver>
                                          </ResizeObserver>
                                        </span>
                                      </Avatar>
                                    </div>
                                  </div>
                                </Col>
                              </ForwardRef(InternalItem)>
                            </div>
                          </div>
                        </Row>
                      </div>
                    </div>
                  </Spin>
                </Spin>
                <div
                  className="ant-list-pagination"
                >
                  <Pagination
                    current={1}
                    onChange={[Function]}
                    onShowSizeChange={[Function]}
                    pageSize={8}
                    total={2}
                  >
                    <LocaleReceiver
                      componentName="Pagination"
                      defaultLocale={
                        Object {
                          "items_per_page": "/ page",
                          "jump_to": "Go to",
                          "jump_to_confirm": "confirm",
                          "next_3": "Next 3 Pages",
                          "next_5": "Next 5 Pages",
                          "next_page": "Next Page",
                          "page": "Page",
                          "page_size": "Page Size",
                          "prev_3": "Previous 3 Pages",
                          "prev_5": "Previous 5 Pages",
                          "prev_page": "Previous Page",
                        }
                      }
                    >
                      <Pagination
                        className=""
                        current={1}
                        defaultCurrent={1}
                        defaultPageSize={10}
                        hideOnSinglePage={false}
                        itemRender={[Function]}
                        jumpNextIcon={
                          <a
                            className="ant-pagination-item-link"
                          >
                            <div
                              className="ant-pagination-item-container"
                            >
                              <ForwardRef(DoubleRightOutlined)
                                className="ant-pagination-item-link-icon"
                              />
                              <span
                                className="ant-pagination-item-ellipsis"
                              >
                                •••
                              </span>
                            </div>
                          </a>
                        }
                        jumpPrevIcon={
                          <a
                            className="ant-pagination-item-link"
                          >
                            <div
                              className="ant-pagination-item-container"
                            >
                              <ForwardRef(DoubleLeftOutlined)
                                className="ant-pagination-item-link-icon"
                              />
                              <span
                                className="ant-pagination-item-ellipsis"
                              >
                                •••
                              </span>
                            </div>
                          </a>
                        }
                        locale={
                          Object {
                            "items_per_page": "/ page",
                            "jump_to": "Go to",
                            "jump_to_confirm": "confirm",
                            "next_3": "Next 3 Pages",
                            "next_5": "Next 5 Pages",
                            "next_page": "Next Page",
                            "page": "Page",
                            "page_size": "Page Size",
                            "prev_3": "Previous 3 Pages",
                            "prev_5": "Previous 5 Pages",
                            "prev_page": "Previous Page",
                          }
                        }
                        nextIcon={
                          <button
                            className="ant-pagination-item-link"
                            tabIndex={-1}
                            type="button"
                          >
                            <ForwardRef(RightOutlined) />
                          </button>
                        }
                        onChange={[Function]}
                        onShowSizeChange={[Function]}
                        pageSize={8}
                        prefixCls="ant-pagination"
                        prevIcon={
                          <button
                            className="ant-pagination-item-link"
                            tabIndex={-1}
                            type="button"
                          >
                            <ForwardRef(LeftOutlined) />
                          </button>
                        }
                        selectComponentClass={[Function]}
                        selectPrefixCls="ant-select"
                        showLessItems={false}
                        showPrevNextJumpers={true}
                        showQuickJumper={false}
                        showTitle={true}
                        style={Object {}}
                        total={2}
                        totalBoundaryShowSizeChanger={50}
                      >
                        <ul
                          className="ant-pagination"
                          style={Object {}}
                        >
                          <li
                            aria-disabled={true}
                            className="ant-pagination-prev ant-pagination-disabled"
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            tabIndex={null}
                            title="Previous Page"
                          >
                            <button
                              className="ant-pagination-item-link"
                              disabled={true}
                              tabIndex={-1}
                              type="button"
                            >
                              <ForwardRef(LeftOutlined)>
                                <AntdIcon
                                  icon={
                                    Object {
                                      "icon": Object {
                                        "attrs": Object {
                                          "focusable": "false",
                                          "viewBox": "64 64 896 896",
                                        },
                                        "children": Array [
                                          Object {
                                            "attrs": Object {
                                              "d": "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z",
                                            },
                                            "tag": "path",
                                          },
                                        ],
                                        "tag": "svg",
                                      },
                                      "name": "left",
                                      "theme": "outlined",
                                    }
                                  }
                                >
                                  <span
                                    aria-label="left"
                                    className="anticon anticon-left"
                                    role="img"
                                  >
                                    <IconReact
                                      icon={
                                        Object {
                                          "icon": Object {
                                            "attrs": Object {
                                              "focusable": "false",
                                              "viewBox": "64 64 896 896",
                                            },
                                            "children": Array [
                                              Object {
                                                "attrs": Object {
                                                  "d": "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z",
                                                },
                                                "tag": "path",
                                              },
                                            ],
                                            "tag": "svg",
                                          },
                                          "name": "left",
                                          "theme": "outlined",
                                        }
                                      }
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="left"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        key="svg-left"
                                        viewBox="64 64 896 896"
                                        width="1em"
                                      >
                                        <path
                                          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                                          key="svg-left-svg-0"
                                        />
                                      </svg>
                                    </IconReact>
                                  </span>
                                </AntdIcon>
                              </ForwardRef(LeftOutlined)>
                            </button>
                          </li>
                          <Pager
                            active={true}
                            itemRender={[Function]}
                            key="1"
                            locale={
                              Object {
                                "items_per_page": "/ page",
                                "jump_to": "Go to",
                                "jump_to_confirm": "confirm",
                                "next_3": "Next 3 Pages",
                                "next_5": "Next 5 Pages",
                                "next_page": "Next Page",
                                "page": "Page",
                                "page_size": "Page Size",
                                "prev_3": "Previous 3 Pages",
                                "prev_5": "Previous 5 Pages",
                                "prev_page": "Previous Page",
                              }
                            }
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            page={1}
                            rootPrefixCls="ant-pagination"
                            showTitle={true}
                          >
                            <li
                              className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                              onClick={[Function]}
                              onKeyPress={[Function]}
                              tabIndex="0"
                              title={1}
                            >
                              <a
                                rel="nofollow"
                              >
                                1
                              </a>
                            </li>
                          </Pager>
                          <li
                            aria-disabled={true}
                            className="ant-pagination-next ant-pagination-disabled"
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            tabIndex={null}
                            title="Next Page"
                          >
                            <button
                              className="ant-pagination-item-link"
                              disabled={true}
                              tabIndex={-1}
                              type="button"
                            >
                              <ForwardRef(RightOutlined)>
                                <AntdIcon
                                  icon={
                                    Object {
                                      "icon": Object {
                                        "attrs": Object {
                                          "focusable": "false",
                                          "viewBox": "64 64 896 896",
                                        },
                                        "children": Array [
                                          Object {
                                            "attrs": Object {
                                              "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                                            },
                                            "tag": "path",
                                          },
                                        ],
                                        "tag": "svg",
                                      },
                                      "name": "right",
                                      "theme": "outlined",
                                    }
                                  }
                                >
                                  <span
                                    aria-label="right"
                                    className="anticon anticon-right"
                                    role="img"
                                  >
                                    <IconReact
                                      icon={
                                        Object {
                                          "icon": Object {
                                            "attrs": Object {
                                              "focusable": "false",
                                              "viewBox": "64 64 896 896",
                                            },
                                            "children": Array [
                                              Object {
                                                "attrs": Object {
                                                  "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                                                },
                                                "tag": "path",
                                              },
                                            ],
                                            "tag": "svg",
                                          },
                                          "name": "right",
                                          "theme": "outlined",
                                        }
                                      }
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="right"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        key="svg-right"
                                        viewBox="64 64 896 896"
                                        width="1em"
                                      >
                                        <path
                                          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                                          key="svg-right-svg-0"
                                        />
                                      </svg>
                                    </IconReact>
                                  </span>
                                </AntdIcon>
                              </ForwardRef(RightOutlined)>
                            </button>
                          </li>
                          <Options
                            changeSize={null}
                            current={1}
                            goButton={false}
                            locale={
                              Object {
                                "items_per_page": "/ page",
                                "jump_to": "Go to",
                                "jump_to_confirm": "confirm",
                                "next_3": "Next 3 Pages",
                                "next_5": "Next 5 Pages",
                                "next_page": "Next Page",
                                "page": "Page",
                                "page_size": "Page Size",
                                "prev_3": "Previous 3 Pages",
                                "prev_5": "Previous 5 Pages",
                                "prev_page": "Previous Page",
                              }
                            }
                            pageSize={8}
                            pageSizeOptions={
                              Array [
                                "10",
                                "20",
                                "50",
                                "100",
                              ]
                            }
                            quickGo={null}
                            rootPrefixCls="ant-pagination"
                            selectComponentClass={[Function]}
                            selectPrefixCls="ant-select"
                          />
                        </ul>
                      </Pagination>
                    </LocaleReceiver>
                  </Pagination>
                </div>
              </div>
            </List>
          </div>
        </Item>
      </div>
    </Space>
  </MediaList>
</Provider>
`;
